File: playground/src/main/kotlin/kr/alham/playground/domain/item/Item.kt
package kr.alham.playground.domain.item

interface Item {
    var id: Long?
    var name: String
    var description: String
    var type: ItemType
    var itemRarity: ItemRarity
    var itemImg: String
}


enum class ItemType {
    EQUIPMENT,
    MATERIAL,
}

enum class ItemRarity(val value: Int) {
    COMMON(1),
    UNCOMMON(2),
    RARE(4),
    EPIC(8),
    UNIQUE(16),
    LEGENDARY(32)
}

File: playground_front/src/types/item.ts
export const ItemTypes = [
    "MATERIAL",
    "EQUIPMENT"
] as const;

export type ItemType = typeof ItemTypes[number];

export const ItemRarities = [
    "COMMON",
    "UNCOMMON",
    "RARE",
    "EPIC",
    "UNIQUE",
    "LEGENDARY"
] as const;

export type ItemRarity = typeof ItemRarities[number];

export const EquipmentTypes = [
    "WEAPON"
] as const;

export type EquipmentType = typeof EquipmentTypes[number];

